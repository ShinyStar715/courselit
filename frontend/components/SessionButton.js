/**
 * This component provides a clickable button which shows if the user
 * is logged in or is a guest.
 */

import React from 'react'
import Link from 'next/link'
import { connect } from 'react-redux'
import {
  CREATOR_AREA_LINK_TEXT,
  GENERIC_SIGNOUT_TEXT,
  GENERIC_SIGNIN_TEXT
} from '../config/strings.js'
import {
  authProps,
  profileProps
} from '../types.js'
import { Grid } from '@material-ui/core'

SessionButton.propTypes = {
  auth: authProps,
  profile: profileProps
}

function SessionButton (props) {
  // const button = props.auth.guest
  //   ? (
  //     <Link href='/login'>
  //       <a>{ GENERIC_SIGNIN_TEXT }</a>
  //     </Link>
  //   ) : (
  //     <Link href='/logout'>
  //       <a>{ GENERIC_SIGNOUT_TEXT }</a>
  //     </Link>
  //   )

  return (
    <Grid container justify='flex-end' spacing={1}>
      {/* {props.profile.isCreator &&
        <Grid item>
          <a href='/create'>{CREATOR_AREA_LINK_TEXT}</a>
        </Grid>
      } */}
      {props.auth.guest
        ? (
          <Grid item>
            <Link href='/login'>
              <a>{ GENERIC_SIGNIN_TEXT }</a>
            </Link>
          </Grid>
        ) : (
          <Grid item>
            <Link href='/logout'>
              <a>{ GENERIC_SIGNOUT_TEXT }</a>
            </Link>
          </Grid>
        )}
    </Grid>
  )
}

const mapStateToProps = state => ({
  auth: state.auth,
  profile: state.profile
})

export default connect(
  mapStateToProps
)(SessionButton)
