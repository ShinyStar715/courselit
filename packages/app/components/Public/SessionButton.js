/**
 * This component provides a clickable button which shows if the user
 * is logged in or is a guest.
 */

import React from "react";
import Link from "next/link";
import { connect } from "react-redux";
import {
  GENERIC_SIGNOUT_TEXT,
  GENERIC_SIGNIN_TEXT
} from "../../config/strings.js";
import { authProps, profileProps } from "../../types.js";
import { Button } from "@material-ui/core";
import { makeStyles } from "@material-ui/styles";

const useStyles = makeStyles({
  button: {
    color: "white"
  }
});

function SessionButton(props) {
  const classes = useStyles();

  return (
    <>
      {props.auth.guest ? (
        <Link href="/login">
          <Button className={classes.button}>{GENERIC_SIGNIN_TEXT}</Button>
        </Link>
      ) : (
        <Link href="/logout">
          <Button className={classes.button}>{GENERIC_SIGNOUT_TEXT}</Button>
        </Link>
      )}
    </>
  );
}

SessionButton.propTypes = {
  auth: authProps,
  profile: profileProps
};

const mapStateToProps = state => ({
  auth: state.auth,
  profile: state.profile
});

export default connect(mapStateToProps)(SessionButton);
