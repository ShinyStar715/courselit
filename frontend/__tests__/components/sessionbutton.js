import React from 'react'
import renderer from 'react-test-renderer'
import SessionButton from '../../components/sessionbutton.js'

test('When user is logged out', () => {
  const component = renderer.create(
    <SessionButton auth={{guest: true, token: ''}} />
  )

  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

test('When user is logged in', () => {
  const component = renderer.create(
    <SessionButton auth={{guest: false, token: ''}} />
  )

  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
