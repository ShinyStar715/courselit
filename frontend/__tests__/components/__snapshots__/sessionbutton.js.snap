// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`When user is logged in 1`] = `
<div>
  <a
    href="/logout"
    onClick={[Function]}
  >
    Logout
  </a>
</div>
`;

exports[`When user is logged out 1`] = `
<div>
  <a
    href="/login"
    onClick={[Function]}
  >
    Login
  </a>
</div>
`;
